### PR TITLE
Fix heroes transition when initial route is not '/'

### PR DESCRIPTION
**Pull request by dnfield**
The current implementation for heroes transitions throws an assertion when a non-default initial route is set, which prints an error message to the console that is likely to confuse users.  The animation ends up being completed by https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/pages.dart#L41-L42

This updates the logic to add another early return for that case, plus a test.

I discovered this when working on the fix for https://github.com/flutter/flutter/issues/23736

/cc @HansMuller (fyi)

### CODE REVIEW

File: packages/flutter/test/widgets/heroes_test.dart [hunk 1]
```diff
@@ -1437,4 +1437,12 @@ void main() {
     expect(find.byKey(firstKey), isInCard);
     expect(find.byKey(secondKey), findsNothing);
   });
+
+  testWidgets('Handles transitions when a non-default initial route is set', (WidgetTester tester) async {
```
[the remark this answers is not in the document]
Comment (dnfield) [reply packages/flutter/test/widgets/heroes_test.dart]: It does fail - MaterialApp creates a heroes for transitions behind the scenes

File: packages/flutter/lib/src/widgets/heroes.dart [hunk 2]
```diff
@@ -598,9 +598,18 @@ class HeroController extends NavigatorObserver {
       final PageRoute<dynamic> to = toRoute;
       final Animation<double> animation = (flightType == HeroFlightDirection.push) ? to.animation : from.animation;
 
-      // A user gesture may have already completed the pop.
-      if (flightType == HeroFlightDirection.pop && animation.status == AnimationStatus.dismissed) {
-        return;
+      // A user gesture may have already completed the pop, or we might be the initial route
+      switch(flightType) {
+        case HeroFlightDirection.pop:
```
[the remark this answers is not in the document]
Comment (dnfield) [reply packages/flutter/lib/src/widgets/heroes.dart]: Done

**Comment by xster** [comment]
LGTM

File: packages/flutter/test/widgets/heroes_test.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (dnfield) [reply packages/flutter/test/widgets/heroes_test.dart]: (I should add that the failure is just that an AssertionError is thrown, but everything seems fine after that).

**Pull request merged**